foreman/data_refinery_foreman/foreman/management/commands/create_compendia.py

```diff
@@ -73,36 +64,28 @@ def handle(self, *args, **options):
         for it. If not a new job will be dispatched for each organism
         with enough microarray samples except for human and mouse.
         """
+        svd_algorithm = 'ARPACK' # default algorithm to arpack until we decide that ranomized is preferred
+        svd_algorithm_choices = ['ARPACK', 'RANDOMIZED', 'NONE']
+        if options['svd_algorithm'] and options['svd_algorithm'] not in svd_algorithm_choices:
+            raise Exception('Invalid svd_algorithm option provided. Possible values are ' + str(svd_algorithm_choices))
+        else:
+            svd_algorithm = options["svd_algorithm"]
+
+        # only include organisms with QN targets. We'll merge groups later.
+        all_organisms = Organism.objects.all().filter(qn_target__isnull=False)
         if options["organisms"] is None:
-            all_organisms = Organism.objects.exclude(name__in=["HOMO_SAPIENS", "MUS_MUSCULUS"])
+            all_organisms = all_organisms.exclude(name__in=["HOMO_SAPIENS", "MUS_MUSCULUS"])
         else:
             organisms = options["organisms"].upper().replace(" ", "_").split(",")
-            all_organisms = Organism.objects.filter(name__in=organisms)
-
-        # I think we could just use options["quant_sf_only"] but I
-        # wanna make sure that values that are not True do not trigger
-        # a truthy evaluation.
-        quant_sf_only = options["quant_sf_only"] is True
-
-        # default algorithm to arpack until we decide that ranomized is preferred
-        svd_algorithm = 'NONE' if quant_sf_only else 'ARPACK'
-        if options["svd_algorithm"] in ['ARPACK', 'RANDOMIZED', 'NONE']:
-            svd_algorithm = options["svd_algorithm"]
+            all_organisms = all_organisms.filter(name__in=organisms)
```

Comment:
Oh nice catch here. We should have been doing it this way before in the now-removed line #80.